pandas/tests/sparse/test_frame.py

```diff
@@ -1267,6 +1267,32 @@ def test_from_scipy_correct_ordering(spmatrix):
     tm.assert_frame_equal(sdf.to_dense(), expected.to_dense())
 
 
+def test_from_scipy_fillna(spmatrix):
+    # GH 16112
+    tm.skip_if_no_package('scipy')
+
+    arr = np.eye(3)
+    arr[1:, 0] = np.nan
+
+    try:
+        spm = spmatrix(arr)
+        assert spm.dtype == arr.dtype
+    except (TypeError, AssertionError):
+        # If conversion to sparse fails for this spmatrix type and arr.dtype,
+        # then the combination is not currently supported in NumPy, so we
+        # can just skip testing it thoroughly
+        return
+
+    sdf = pd.SparseDataFrame(spm).fillna(-1.0)
+
+    # Returning frame should fill all nan values with -1.0
+    expected = pd.SparseDataFrame([[1, -1, -1],
+                                   [-1, 1, -1],
+                                   [-1, -1, 1.]])
+
+    tm.assert_numpy_array_equal(sdf.values, expected.values)
```

Comment:
Indeed, SparseSeries `{0: nan, 1: 1, 2: nan}` can have the underlying SparseArray of shape `(3,)` (`[nan, 1, nan]`, sparse block of length 3, start index 0) or of shape `(1,)` (`[1]`, sparse block of length 1, start index 1).

I did manage something.